pkg/admissioncontroller/server/handlers/webhooks/utils.go

```diff
@@ -56,3 +61,43 @@ func respond(w http.ResponseWriter, response *admissionv1beta1.AdmissionResponse
 		logger.Logger.Error(err)
 	}
 }
+
+// DecodeAdmissionRequest decodes the given http request into an admission request.
+// An error is returned if the request exceeds the given limit.
+func DecodeAdmissionRequest(r *http.Request, decoder runtime.Decoder, into *admissionv1beta1.AdmissionReview, limit int64) error {
+	// Read HTTP request body into variable.
+	var (
+		body              []byte
+		wantedContentType = runtime.ContentTypeJSON
+		// Increase limit by 1 (spare capacity) to determine if the limit was exceeded or right on the mark after reading.
+		lr = &io.LimitedReader{R: r.Body, N: limit + 1}
+	)
+
+	if r.Body != nil {
+		data, err := ioutil.ReadAll(lr)
+		if err != nil {
+			return err
+		}
+		if lr.N <= 0 {
+			return apierrors.NewRequestEntityTooLargeError(fmt.Sprintf("limit is %d", limit))
+		}
+		body = data
+	}
+
+	// Verify that the correct content-type header has been sent.
+	if contentType := r.Header.Get("Content-Type"); contentType != wantedContentType {
```

Comment:
Hmm, does that mean that the webhook will fail if someone is trying to create a Shoot with `application/vnd.kubernetes.protobuf`?

Reply:
I don't think so because the API server sends the request to webhook (https://github.com/kubernetes/kubernetes/blob/e36e68f5f6964a3e612b36bfea0bb17c8f05e083/staging/src/k8s.io/apiserver/pkg/util/webhook/client.go#L150), not the client.